### PR TITLE
Move ValidTargetFilename dependencies to libtuf

### PR DESCRIPTION
**Pull request by jochenschneider**
Also remove some superdfluous imports.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -53,7 +53,7 @@ lazy val serverDependencies = libraryDependencies ++= {
 lazy val commonSettings = Seq(
   organization := "com.advancedtelematic",
   scalaVersion := "2.12.4",
-  scalacOptions := Seq("-unchecked", "-deprecation", "-encoding", "utf8", "-Xexperimental", "-Ypartial-unification"),
+  scalacOptions := Seq("-unchecked", "-deprecation", "-encoding", "utf8", "-Xexperimental", "-Ypartial-unification", "-Ywarn-unused-import"),
```
Comment (simao) [inline build.sbt]: I don't think we should add this flag. The ide should do this for you and it's a pain in the ass in development seeing all those warnings, leads users to just ignore all warnings.

File: libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/data/Messages.scala [hunk 2]
```diff
@@ -29,4 +30,28 @@ object Messages {
...
   implicit val packageStorageUsageMessageLike = MessageLike[PackageStorageUsage](_.namespace)
+
+  case class DeviceUpdateReport(namespace: Namespace, device: DeviceId, updateId: UpdateId, timestampVersion: Int,
+                                operationResult: Map[EcuSerial, OperationResult], resultCode: Int)
+
+  implicit val operationResultEncoder: Encoder[OperationResult] = deriveEncoder
+  implicit val operationResultDecoder: Decoder[OperationResult] = deriveDecoder
+
+  implicit val deviceUpdateReportEncoder: Encoder[DeviceUpdateReport] = deriveEncoder
+  // For backwards compatibility reasons we have a decoder that can parse DeviceUpdateReport
```
Comment (simao) [inline libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/data/Messages.scala]: when do we need this? Is this still the case?

File: build.sbt (hunk 1 above)
Comment (jochenschneider) [reply build.sbt]: ok, removed

File: libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/data/Messages.scala (hunk 2 above)
Comment (jochenschneider) [reply libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/data/Messages.scala]: probably not, simplified

**Comment by jochenschneider** [comment]
Updated.

File: build.sbt (hunk 1 above)
Comment (simao) [reply build.sbt]: you can leave it on if it helps you. I can remove it when developing and add it back before commit.

**Comment by simao** [comment]
This looks good but please do not merge it before #166 Conflicts would be easier to solve here because it's a smaller PR and the other one is quite sensitive and @jerrytrieu is already testing it.

**Comment by jochenschneider** [comment]
Ok.

File: build.sbt (hunk 1 above)
Comment (jochenschneider) [reply build.sbt]: Can this be somehow set in the user config?

**Pull request merged**